src/Crawler/MigrateCrawlObserver.php

```diff
@@ -131,7 +120,7 @@ public function crawled(
 
         $cacheUrl = $url_string;
         // $url instanceof UriInterface ? $url->__toString() : null;
-        $cacheFoundOnUrl = $foundOnUrl instanceof UriInterface ? $foundOnUrl->__toString() : null;
+        $cacheFoundOnUrl = $foundOnUrl instanceof UriInterface ? $foundOnUrl->__toString() : $url_string;
```

Comment:
I'm fairly sure this is a bug fix - the cached object was getting

```
'foundOnUrl' => null
```

Which caused issues with creating URI instances from that, this will just default to the current hit URL if the `foundOnUrl` is not passed in directly.